### PR TITLE
[1.x] Add option to force the password to have a special character.

### PR DESCRIPTION
**Pull request by leo95batista**
Hi! I have created this PR with the intention of adding in the password validation rules the option to require that the password contain or not special characters. 

Regards.

### CODE REVIEW

**Comment by DarthShmev** [comment]
Looks good to me.

File: src/Rules/Password.php [hunk 1]
```diff
@@ -66,14 +77,39 @@ public function message()
...
-            return __('The :attribute must be at least '.$this->length.' characters and contain at least one number.');
-        } elseif ($this->requireUppercase && $this->requireNumeric) {
-            return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character and number.');
-        } else {
-            return __('The :attribute must be at least '.$this->length.' characters.');
+        switch (1) {
+            case $this->requireUppercase
+                && ! $this->requireNumeric
+                && ! $this->requireSpecialCharacter:
+                return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character.');
```
Comment (buismaarten) [inline src/Rules/Password.php:84]: It should probably changed to something like: `return __('The :attribute must be at least :length characters and contain at least one uppercase character.', ['length' => $this->length]);`

**Comment by taylorotwell** [comment]
This PR has merge conflicts. Also why `switch(1)` instead of `switch(true)`?

**Comment by leo95batista** [comment]
Hello, good morning everyone! , I am reviewing, and I see that the PR #70 was approved first before this PR #65. For that reason the conflict.  In PR #70, they changed the style in which the ```validation message was returned```. When creating my PR, I can change the style, but I did not do it to maintain the style that existed initially.


**Before**
```
return __('The :attribute must be at least '.$this->length.' characters and contain at least one uppercase character.');
```

**Now**
```
return __('The :attribute must be at least :length characters and contain at least one uppercase character.', ['length' => $this->length]);
```

I think ```switch(1)``` and ```switch(true)``` mean the same in that context

File: src/Rules/Password.php (hunk 1 above)
Comment (leo95batista) [reply src/Rules/Password.php]: Please check it now

Comment (buismaarten) [reply src/Rules/Password.php]: @leo95batista looks better!

**Comment by leo95batista** [comment]
@taylorotwell , do you think anything else needs to be changed before doing the merge?

**Pull request merged**